ably/realtime_conn.go

```diff
@@ -90,6 +94,9 @@ func (c *Conn) reconnect(result bool) (Result, error) {
 	connKey := c.details.ConnectionKey
 	connSerial := c.serial
 	c.state.Unlock()
+	defer func() {
```

Comment:
Why the defer? Please add a comment if there's a reason for it, otherwise remove it.

Plus, it's not true we're reconnecting if `c.connectWithRecovery` returns an error.

Reply:
Done at ca27f152968950cd4bb5ecab15a9b33aea75d5c3.